effekt/shared/src/main/scala/effekt/Lexer.scala

```diff
@@ -169,7 +169,7 @@ class Lexer(source: Source) {
    */
   var current: Int = 0
   /** The sequence of tokens the source strings contains. Returned as the result by [[Lexer.run()]] */
-  val tokens: mutable.ListBuffer[Token] = mutable.ListBuffer.empty
+  val tokens: mutable.ArrayBuffer[Token] = new mutable.ArrayBuffer(initialSize = 256)
```

Comment:
I think this is the main culprit, actually: we know we'll have enough tokens to warrant reserving a bit of space. :)